[rpm_key] Pass GPG key as string value

infame-io opened the pull request:
##### SUMMARY

<!--- Describe the change below, including rationale and design decisions -->
It would be great if the GPG key could be passed as a string value into the `ansible.builtin.rpm_key` module. This would allow passing the GPG key as an env. variable
```bash
export GPG_KEY="-----BEGIN PGP PUBLIC KEY BLOCK-----

mQGNBGUhLi4BDACrXw5kkEdclg7edHjzs3dCs4GIn7ct/tkyhi0Sw5cmH8txujPi
...
6cy9+l5zdaTzJ+biexvyK7dhXXhyueXWaDF9
=NXmg
-----END PGP PUBLIC KEY BLOCK-----"
```

```yaml
---
- name: Import GPG key
  hosts: sandwich

  tasks:
    - name: Import key (env. var)
      become: true
      ansible.builtin.rpm_key:
        key: '{{ lookup("env", "GPG_KEY" )}}'
        state: present
```

and it could enable passing the GPG key from services like AWS Systems Manager Parameter Store
```yaml
---
- name: Import GPG key
  hosts: sandwich

  tasks:
    - name: Import key (AWS SSM)
      become: true
      ansible.builtin.rpm_key:
        key: '{{ lookup("aws_ssm", "/test/ansible/rpm.key.gpg", region="us-east-1" )}}'
        state: present
```
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE

<!--- Pick one below and delete the rest -->

- Feature Pull Request

##### ADDITIONAL INFORMATION

<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

```paste below

```

infame-io committed:
[rpm_key] Add GPG key from a string value

infame-io committed:
[rpm_key] Add GPG key from a string value

bcoca commented on lib/ansible/modules/rpm_key.py:
```diff
@@ -116,6 +116,10 @@ def __init__(self, module):
             keyid = self.getkeyid(keyfile)
             should_cleanup_keyfile = True
+        elif is_pubkey(key):
+            keyfile = self.keyfile(bytes(key, 'utf-8'))
```
use ``to_bytes`` function instead

infame-io replied:
Updated

bcoca commented on lib/ansible/modules/rpm_key.py:
```diff
@@ -148,6 +152,15 @@ def __init__(self, module):
                 module.exit_json(changed=False)
 
+    def keyfile(self, key):
+        tmpfd, tmpname = tempfile.mkstemp()
```
the module already has a private tmpdir, use that to keep files in a place we guarantee cleanup

bcoca reviewed (changes requested):
also missing a changelog

infame-io committed:
[rpm_key] Add GPG key from a string value

infame-io commented:
Thank you @bcoca. Changelog included

bcoca reviewed (changes requested):
one thing I missed, the `key:`  option is missing an update, add 'since version 2.17 you can also pass the key as a string'

infame-io committed:
[rpm_key] Expanding 'key' description

artificial-intelligence committed:
password_hash: update docs about bcrypt (#81675)

Signed-off-by: Sven Kieske <kieske@osism.tech>

mishrahrishikesh committed:
Update distro version in __init__.py (#83114)

Update the bundled package distro from 1.6.0 to 1.9.0 in __init__.py

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
inspect components, ansible_managed templatable (#83053)

* inspect components, ansible_managed templatable

fixes #82322

bcoca committed:
removed extra ansible_managed

moreati committed:
Add description of ansible.utils.path.unfrackpath() basedir argument

s-hertel committed:
Fix installing roles containing symlinks (#82911)

* Fix installing roles containing symlinks

Fix sanitizing tarfile symlinks relative to the link directory instead of the archive

For example:

role
├── handlers
│   └── utils.yml -> ../tasks/utils/suite.yml

The link ../tasks/utils/suite.yml will resolve to a path outside of the link's directory, but within the role

role/handlers/../tasks/utils/suite.yml

the resolved path relative to the role is tasks/utils/suite.yml, but if the symlink is set to that value, tarfile would extract it from role/handlers/tasks/utils/suite.yml

* Replace overly forgiving test case with tests for a symlink in a subdirectory of the archive and a symlink in the archive dir when these are not equivalent.

* Build test case from role files to make it easier to add test cases

Fixes #82702
Fixes #81965
Fixes #82051

Akasurde committed:
assemble: update argument_spec with 'decrypt' option

* decrypt option is used by assemble action plugin.
  Add this parameter to remove failure raised by
  validate-modules:nonexistent-parameter-documented

Fixes: #80840

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
distro: remove pep8 ignore

* Remove unnecessary pep8 from ignore.txt

Fixes: #80840

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

oraNod committed:
minor fix for misspelling (#83156)

This change fixes a minor spelling error, indivudually ==> individually.

jborean93 committed:
Add version ceiling for pypsrp (#83122)

* Add version ceiling for pypsrp

Add a version ceiling for the pypsrp requirements. This ensures that the
future v1.0.0 release won't impact existing users who install the
library using the version range specified.

* Use constraints file

* Add changelog for this change

felixfontein committed:
Correctly treat separate paragraphs in top-level descriptions as separate paragraphs. (#83155)

felixfontein committed:
Show examples in role entrypoint argument specs. (#82671)

sivel committed:
Gather sysinfo facts on s390 (#83157)

Akasurde committed:
uri: deprecate 'yes' or 'no' value in follow_redirects param (#83169)

* uri (module and lookup): deprecate 'yes' or 'no' value in follow_redirects param

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sivel committed:
Don't hardcode the dnf module, dynamically select one (#83183)

bcoca committed:
package_facts add alias support (#83149)

* package_facts add alias support

fixes #83143

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

nitzmahone committed:
clean up ansible-connection (#82992)

* clean up ansible-connection stuff

* eliminate unnecessary usage of pty/termios
* always use default pickle protocol
* remove unnecessary wire hashing

Co-authored-by: Kate Case <this.is@katherineca.se>

bcoca committed:
display: fix log verbosity (#83179)

* display, ensure we have a logger before we log

Akasurde committed:
git: Check if version is populated or not (#82173)

Before usage check if the git version is populated or not.

Fixes: #72321

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Build test modules for aarch64 (#83195)

mattclay committed:
Multi-arch support for container test (#83196)

j-carl committed:
Fix order of CLI arguments on FreeBSD (#81377)

When passing arguments on FreeBSD to service command the order is
important, otherwise the service command will ignore the arguments.

flowerysong committed:
ansible-doc: fix typo in output (#83205)

bcoca committed:
ansible-galaxy, clarify licensing for 'init' templates and results (#83175)

lostsoulparty committed:
facts: Update RAID devices regex for FreeBSD (#82081)

Added support FreeBSD RAID devices and regex to match partitions as well as slices. 
RAID device list is taken from here: https://github.com/freebsd/freebsd-src/blob/main/usr.sbin/bsdconfig/share/device.subr

Akasurde committed:
pymarkdown: update docs (#83150)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
added examples for systemd service usage (#83192)

mostly how to filter/manipulate the results
---------

Co-authored-by: flowerysong <junk+github@flowerysong.com>

geckoflume committed:
package_facts: apk fix when cache is empty (#83132)

mkrizek committed:
dnf: integration tests cleanup (#83203)

* remove no longer needed tasks
* remove workarounds
* consolidate conditionals
* remove distros/versions from conditionals that are no longer present
  in CI

Akasurde committed:
Darwin: add unit tests for hardware fact gathering (#83212)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
dnf5: implement enable_plugin and disable_plugin options (#83105)

https://github.com/rpm-software-management/dnf5/commit/80cfea9c2514704058ce501c496433fbb6e349bf

mkrizek committed:
Refactor yum_repository, deprecate yum only options (#83116)

Akasurde committed:
Use inclusive word: logical (#83232)

Use word 'logical' instead of non-inclusive word

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

s-hertel committed:
remove old install info when installing collections (#83194)

* Fix cleaning up galaxy server info from previous installations

* changelog

Akasurde committed:
Refactor dmidecode fact gathering code (#83211)

* Code now bails out early if dmidecode command is unavailable

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
setup module, make missing sysctl minor issue (#81297)

Signed-off-by: Brian Coca <brian.coca+git@gmail.com>
Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>
Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Warn if the binary is unavailable using get_bin_path (#83258)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
Remove deprecated VariableManager._get_delegated_vars (#83259)

Fixes #82950

sourn00dl committed:
MUNI tech writers: timeout documentation. (#83152)

* Documentation on the timeout keyword

* wording changes

* keyword description wording edit

* removed whitespace at the end of lines.

* Changed to more accurate wording

---------

Co-authored-by: Niko Mlynarčik <536617@mail.muni.cz>

mattclay committed:
ansible-test - Update ansible-test-utility-container (#83274)

mattclay committed:
ansible-test - Update pypi-test-container (#83273)

mattclay committed:
ansible-test - Update http-test-container (#83272)

* Disable cipher suite integration tests

When the client and server both support TLS 1.3, cipher suite selection is effectively a no-op.

Python provides no way to disable TLS 1.3 cipher suites, and no way to prefer user-selected cipher suites.
Since TLS 1.3 includes a mandatory cipher suite, this makes it impossible to induce a cipher suite related error when using TLS 1.3.

* Update multipart form-data integration test

* Fix netrc integration tests

* Update HTTP test host substring

* ansible-test - Update http-test-container

appeltel committed:
Correct code comment in PBE to reflect current behavior (#83269)

Correct a code comment in the PlaybookExecutor class that explains
the host failure condition under which the PBE breaks out of the
run. This comment previously stated that the maximum failure
percentage was checked, but that logic has been refactored into
the linear strategy plugin.

Akasurde committed:
filter: update the documentation for win_splitdrive (#83301)

Fixes: #83299

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
apt_*: add more info message while updating apt cache (#83230)

Fixes: #77941

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
get_url: update docs for checksum (#83305)

If user provides checksum URL which requires basic auth,
then module parameter username and password is used to
download the checksum file.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
ansible-doc: make color configurable (#83311)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
config, integrate dynamic galaxy servers (#83129)


Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

bcoca committed:
gather_facts, mention ansible_facts_parallel var (#83278)

odra committed:
file module, follow symlink when doing hardlink (#34228)

Fixes: #33911

s-hertel committed:
Fix task.resolved_action callbacks (#82003)

* Fix task.resolved_action for callbacks when playbooks use action or local_action

* Fix using module_defaults with 'action' and 'local_action' task FA and add a test case

Fixes #81905

Akasurde committed:
test: Cleanup facts/network/* tests (#83256)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

b-enoit-be committed:
The extension returned by the splitext contains the dot (#82803)

Fixed examples for splitext filter plugin

garciart committed:
Documented v2_runner_on_failed method in Callback class (#83267)

* Documented v2_runner_on_failed method in Callback class

* Added changes requested by @bcoca.

jborean93 committed:
powershell - Improve Add-Type tempdir handler (#83080)

Improves the Add-Type temporary directory handler to include a retry
mechanism and not fail on an error. Deleting a temporary file used in
compilation is not a critical error and should improve the reliability
of Ansible on Windows hosts.

yctomwang committed:
Fix Test failure with cowsay installed/present (#83347)

bcoca committed:
timeout give extra info (#83206)

the new field shows the python code in execution when it timed out, 99% of the time it will be on a selector waiting for output from ssh to remote.

bcoca committed:
additional timeout info (#83364)

Akasurde committed:
facts: add facts about x86_64 flags to detect microarchitecture (#83356)

Fixes: #83331

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
bool filter, add booleanization strategy option (#83362)

 also make the YAML booleanization the same as spec/JSON/module function
 previous 'aproximation' was missing several options.

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

russoz committed:
docs adjustments (batch 2) (#83368)

* docs adjustments

* Update lib/ansible/modules/assemble.py

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

---------

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

s-hertel committed:
fix any_errors_fatal tests (#83389)

* fix any_errors_fatal test to exit on non-zero rc

Use a typo in the debug msg to avoid matching play recap

* remove duplicate 'set -ux'

sivel committed:
Properly quote all needed components of shell commands (#83365)

* Properly quote all needed components of shell commands

* Use self.quote, add new self.join

s-hertel committed:
Revert "bool filter, add booleanization strategy option (#83362)" (#83391)

This reverts commit 655a8ff38f83dbc248a80cb4b79f67226674c58f.

mattclay committed:
Update azure-pipelines-test-container (#83302)

mkrizek committed:
Validate and process Handler.listen only once (#83400)

Fixes #83392

mattclay committed:
Fix quoting in fork_safe_stdio test (#83412)

mattclay committed:
Fix issues reported by mypy (#83413)

russoz committed:
docs adjustments (#83369)

Akasurde committed:
Remove extraneous pylint ignore (#83257)

* Pylint issue 511 is closed, remove ignore related to that.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
ansible-test - Prepare for Python 3.13 (#83414)

Includes an update to `coverage` and pinned Python requirements.

russoz committed:
docs adjustments - batch(ghi) (#83399)

* review modules docs - batch(ghi)

* fix option markup in gather_facts

flowerysong committed:
Clarify documentation for `INJECT_FACTS_AS_VARS` (#83404)

sivel committed:
Specify -O in windows ssh tests as needed for scp to work with openssh 9+ (#83426)

mattclay committed:
Add Python 3.13 support (#83416)

* Add Python 3.13 support

* Add temporary work-around for yamllint test

Akasurde committed:
Revert warning message in setup/gather_facts (#83427)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jborean93 committed:
ansible-test action-plugin-docs sidecar (#83325)

Fix to have ansible-test sanity --test action-plugin-docs to check for
action plugin documentation inside a sidecar file rather than a Python
module.

lukecdavidson committed:
Fix Linux Sector count Fact (#81246)

* Use sectorsize fact in disk size calc rather than hardcoded 512

* Added conversion for sysfs sectorcount to real sector count

mattclay committed:
ansible-test - Fix unit test coverage collection (#83433)

mattclay committed:
ansible-test - Update PyPI test container to 3.1.0 (#83432)

russoz committed:
review modules docs - batch(klmpr) (#83418)

colin-nolan committed:
Adds `limit` parameter to `ansible.builtin.find`  (#83153)

* Explicitly guarantee topdown search.
* Makes max matches unlimited == None.

gilsongpfe committed:
unarchive: Better handling of files with an invalid timestamp in zip file (#81520)

Fixes: #81092

Signed-off-by: gilsongpfe <gilson.gpf@gmail.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

flowerysong committed:
dnf: update `use_backend` documentation (#83429)

* add note about backwards compatibility

Akasurde committed:
vault: Handle directory value to vault password file (#83384)

When vault password file env variable is set to blank,
this value is converted to CWD and passed for further
processing.
Check if ANSIBLE_VAULT_PASSWORD_FILE is not a directory.

Fixes: #42960

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
Remove Python 3.10 support for the controller (#83221)

Fixes #83094

mattclay committed:
selinux compat - add ignore, proper ex chaining (#83460)

mattclay committed:
Remove timezone support module and tests (#83465)

The timezone support module was used only for changing the timezone in the user module integration tests.

Changing the timezone for the tests is unecessarily complex for the purpose of asserting proper parsing of user expiration times.

garciart committed:
Added docstrings to V2 methods in the CallbackBase Class (2 & 3 of 27) (#83342)

mattclay committed:
ansible-test - Update tested platforms (#83446)

* ansible-test - Update tested platforms

* Add work-around for Alpine tests

* Fix prepare_http_tests on Fedora

* Fix deb-src setup for Ubuntu 24.04+

* Set multiarch_test_pkg for Ubuntu 24.04

* Update AZP test matrix

shiftyphil committed:
service_facts on OpenBSD: Don't crash on '=' in rcctl flags. (#83458)

Splitting on all '=' characters produced too many values.

Akasurde committed:
replace: update after/before example (#83453)

Fixes: #83390

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
Update the dnf5 copr repo (#83459)

The dnf-nightly is the official nightly repo now.

yctomwang committed:
human_to_bytes: strictly parse strings (#83403)

Fixes: #82075

felixfontein committed:
ansible-test: update http-test-container to 3.2.0 (#83469)

mattclay committed:
ansible-test - Update PyPI and utility container (#83475)

dbrgn committed:
get_url: Verify checksum using tmpsrc, not dest (#64092)

Previously, if the checksum of the downloaded file did not match the
specified checksum, the *destination* file was removed. This possibly
leaves the system that is being provisioned in an invalid state.

Instead, the checksum should be calculated on the temporary file only.
If there's a mismatch, delete the *temporary* file, not the destination
file.

This requires checking the checksum before moving the file.

Akasurde committed:
unarchive: handle content and size differences (#83454)

Consider content differs and size differs while unarchiving
the same tar.gz file

Fixes: #29610

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
ansible-test - Replace FreeBSD 14.0 with 14.1 (#83477)

konstruktoid committed:
add support for inactive option (#83355)

Signed-off-by: Thomas Sjögren <konstruktoid@users.noreply.github.com>

bcoca committed:
Ensure the correct connection name is shown in results (#83354)

bcoca committed:
linear strategy, show templated task name on start (#83473)

we only templated in some cases but when queueing we could get an untemplated name for the 'on start' event.

mattclay committed:
Enable Ubuntu 24.04 group 6 in CI (#83466)

* Enable Ubuntu 24.04 group 6 in CI

* Disable rootfull Podman on Ubuntu

* Disable unix-chkpwd AppArmor profile on Ubuntu for Fedora 40 tests

* Document AppArmor and rootfull issues

mkrizek committed:
dnf, dnf5: fix installing via wildcard (#83481)

Fixes #83373

sminux committed:
Fix simmilar branches (#83497)

Identical branches in conditional node: 'elif name != permanent_hostname' and 'else'. First bruch is not nessesary

Co-authored-by: mar.s <marchenkov.s.d@gmail.com>

mkrizek committed:
dnf - arches must be the same in the is_newer_installed check (#83417)

Fixes #83406

BenoitKnecht committed:
facts/hardware: Fix support_discard block device fact (#83480)

Previously, `support_discard` simply returned the value of
`/sys/block/{device}/queue/discard_granularity`. When its value is `0`, then
the block device doesn't support discards; _however_, it being greater than
zero doesn't necessarily mean that the block device _does_ support discards.

But another indication that a block device doesn't support discards is
`/sys/block/{device}/queue/discard_max_hw_bytes` being equal to `0` (with the
same caveat as above). So if either of those are `0`, set `support_discard` to
zero, otherwise set it to the value of `discard_granularity` for backwards
compatibility.

Signed-off-by: Benoît Knecht <bknecht@protonmail.ch>

mkrizek committed:
dnf: remove legacy code (#83298)

mkrizek committed:
Fix tb for when env var contains % (#83499)

Fixes #83498

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

cstiago committed:
Replace references to archived ansible/ansible-examples repository (#83500)

* Replace references to archived ansible/ansible-examples repository

* Update examples to reference ansible repository

* Include HTTPS among SSH references in examples

bcoca committed:
actually show plugin config warnings/deprecations (#82593)

previouslly we recorded but did not show to avoid spam
since we could not dedup from forks, that was already
fixed in another PR so now we can show/display them.

Also:
  * funcitonalize deprecation msg construct from docs
  * reuse formatting func in cli
  * normalize alternatives: most of the code used intended plural
    but some and most data/tests used the singular
  * update schemas and tests

Co-authored-by: Matt Davis <6775756+nitzmahone@users.noreply.github.com>
Co-authored-by: Felix Fontein <felix@fontein.de>

felixfontein committed:
Fix require_only_one. (#83511)

bcoca committed:
hostvars templating fix, override serialization (#83509)

fixes #82872
nicer implementation courtesy of nitzmahone

mkrizek committed:
Fix incorrect rc when executing end_host in rescue (#83522)

Fixes #83447

mkrizek committed:
linear: fix included handlers executing in lockstep (#83209)

Fixes #83019

flowerysong committed:
package_facts: fix warning logic (#83520)

* package_facts: fix warning logic

* Refactor so that warnings can work

felixfontein committed:
validate-modules: reject option/alias names equal up to casing belonging to different options (#83530)

* Reject option/alias names equal up to casing belonging to different options.

* Update test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/main.py

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

yctomwang committed:
Fix Creating user directory using tilde always reports "changed" (#83113)

Fixes: #82490

HVSharma12 committed:
Add SL-Micro to Suse family list (#83541)

jcgruenhage committed:
Add dinit as a service_mgr (#83489)

flowerysong committed:
known_hosts: don't modify `AnsibleModule.params` (#83517)

Using this dictionary to store the return values results in
the return values showing up in the returned
`invocation['module_args']`, which is confusing. It also causes all
module arguments to be returned, which is preserved by this change but
should ideally be removed in the future.

DhaneshRaghavan committed:
update distro version to 1.9.0 (#83540)

russoz committed:
review modules docs - batch(fragments) (#83431)

Co-authored-by: flowerysong <junk+github@flowerysong.com>

mattclay committed:
Quality-of-life improvements for release tool (#83551)

* Quality-of-life improvements for release tool

- Default devel releases to b1
- Default non-devel releases to rc1
- Default to release announcement to console
- Avoid auto-links in GH release annoucements for file sizes

mattclay committed:
ansible-test - Update base/default containers (#83553)

mattclay committed:
ansible-test - Update nios-test-container to 4.0.0 (#83539)

russoz committed:
fix misc. documentation typos and formatting for modules starting with s (#83420)

russoz committed:
fix misc. documentation typos and formatting for modules starting with u-y (#83422)

mkrizek committed:
dnf: follow-up on is_newer_installed arches fix (#83556)

* dnf: follow-up on is_newer_installed arches fix

* fix for the non package object case

* prevent early bailout in dnf _is_newer_version_installed
* non-installed available arches would fail out of the check early

---------

Co-authored-by: Matt Davis <nitzmahone@redhat.com>

rebtoor committed:
Ensure that we are handling DownloadError properly in the dnf module (#83543)

Signed-off-by: Roberto Alfieri <ralfieri@redhat.com>

russoz committed:
remove ignore clauses for module blockinfile (#83577)

bcoca committed:
module compression is now again settable via var (#83573)

Previous change overlooked 'uncommenting' the variable entry as a way to update this  to keep the functionality.


Co-authored-by: Glandos <bugs-github@antipoul.fr>
Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

alex-barth committed:
Update any.yml (#83588)

* fixed typo in description
* fixed comment in examples

mattclay committed:
ansible-test - Improve unknown env error message (#83610)

russoz committed:
replace - fix documentation and remove ignore clauses (#83574)

pabigot committed:
debconf: handle boolean value representation consistently (#83601)

* lift code that normalizes value type for boolean vtype to cover both
  branches of conditional.
* remove obsolete and incomplete conversion of type in set_selection.

Fixes: #83594
Signed-off-by: Peter A. Bigot <pab@pabigot.com>

russoz committed:
remove ignore clauses for module lineinfile (#83595)

webknjaz committed:
🧪 Use `setuptools [core]` @ collections_runtime_pythonpath (#83627)

mattclay committed:
Revert "🧪 Use `setuptools [core]` @ collections_runtime_pythonpath (#83627)"

This reverts commit 0d5460df16af2c8e082796c6a46cb075060f74c9.

mattclay committed:
Use build instead of pep517 for integration test

krescent committed:
env: update examples for setting default values to variable (#83370)

russoz committed:
docs adjustments (batch 1) (#83341)

russoz committed:
remove ignore clauses for module find (#83575)

add testcase for expanding home dir names

mattclay committed:
ansible-test - Remove VyOS tests and support files (#83650)

The VyOS remote image required for running the tests is no longer functional.

teekwan committed:
Added info about MacOS targets for cron module (#83620)

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
display: warn user about empty log path value (#83381)

Fixes: #79959

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

s-hertel committed:
increase timeout downloading galaxy role URLs (#83562)

* Update the timeout to download role archive to 60 seconds to match collections
TODO: in future make a configurable 'artifact download timeout'

Akasurde committed:
apt: Report change when package is removed (#83547)

While upgrade process removes a package, module should
report changed=True instead of changed=False

Fixes: #46314

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Replace random with secrets (#83668)

Use secrets library instead of random.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jborean93 committed:
Remove explicit shell environment docs (#83649)

Removes the docs for the environment keyword in the shell base plugins
as they are a no-op in the plugins themselves. The environment value is
provided by the action base which gets it from the action base on the
task.environment value. This should avoid confusion around how its being
set and removes some code that is not used at all.

Akasurde committed:
Set LANGUAGE env variable is set to a non-English locale (#83671)

Fixes: #83608

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

leegarrett committed:
Typo fixes and other bits and bobs (#83672)


Co-authored-by: Lee Garrett <lgarrett@rocketjump.eu>

felixfontein committed:
validate-modules: detect names set mismatch between argument spec and documentation (#83599)

mkrizek committed:
dnf/setup_rpm_repo: simplify dummy testing repos (#83624)

This change simplifies construction and footprint of testing rpm
repository created by rpmfluff:
* all packages default to noarch
* only when necessary build package for a specific architecture(s)
* do not build several repositories each for specific arch
* remove duplicate "incompatible arch" test
* skip_broken_and_nobest: move testing packages from an external repo
  into our dummy repo for transparency
* remove compatibility code from create_repo.py for versions no longer
  needed
* remove support for old OS versions from setup_rpm_repo
* simplify representation of an rpm in create_repo.py to allow easier
  changes

Notes
* there is one more external testing repo used at
https://ci-files.testing.ansible.com/test/integration/targets/setup_rpm_repo/repo-with-updateinfo

Akasurde committed:
Remove selinux import (#83674)

Remove selinux import which was kept for backward compatibility

Fixes: #83657

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

s-hertel committed:
fix module_defaults group incorrectly giving deprecation warnings (#83510)

don't display deprecation warnings for actions/modules as a result of using an action_group containing a deprecated plugin

jborean93 committed:
winrm - quota retry handling (#83656)

* winrm - quota retry handling

Add a retry attempt when receiving ERROR_WSMAN_QUOTA_MAX_OPERATIONS when
starting a command. This can occur when running a loop with multiple
iterations or an action plugin that runs multiple commands.

* Update pywinrm constraint for test

* Add verbose hint and mark test as destructive

Akasurde committed:
tests: use keyserver with keyid while using apt_key (#83694)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
test: update tests (#83686)

* Remove commented code
* Enable disabled tests
* Formatting

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

karlg100 committed:
add error handling when parsing values in ini files (#82718)

Fixes: #82717

Co-authored-by: Karl A. Grindley <kgrindley@ll.mit.edu>

skupfer committed:
Add UID and GID min/max keys (#81770)

Fixes: #72183

Akasurde committed:
ansible-doc: handle on_fail (#83676)

Handle errors raised when role doc has errors

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
systemd facts: Handle AttributeError (#83684)

* Handle AttributeError raised while running systemd facts on
  non-systemd hosts

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Remove EXTERNALLY-MANAGED marker in apt test (#83706)

The marker is removed in ansible-test managed environments, but the apt test restores it
by installing/upgrading packages. To avoid breaking later tests, the marker needs to be
removed again.

ci_complete

Akasurde committed:
Remove extraneous get_bin_path call (#83675)

ip_path is already calculated before calling get_interfaces_info

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>
Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>
Co-authored-by: Brian Coca <bcoca@users.noreply.github.com>

Akasurde committed:
test: Handle Singleton Display class (#83673)

Fixes: #83538

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
release.py - Auto-update setuptools upper bound (#83713)

When releases are prepared, the upper bound on setuptools in pyproject.toml will be automatically updated
to the latest version available on PyPI. This version will then be tested by the package-data sanity test
during the release process and will be used to build the release.

This change ensures that a released version of ansible-core can be built in the future if a new setuptools
release includes breaking changes that would prevent building a functional package. If a downstream package
maintainer requires a newer setuptools version than the upper bound permits, they can patch pyproject.toml
as needed. Since ansible-core releases support specific Python versions, lack of support for new setuptools
releases will have no effect on support for future Python versions.

mattclay committed:
ansible-test - Update coverage to 7.6.1 (#83723)

mattclay committed:
Fix type hints and misc mypy/pylint issues (#83724)

bcoca committed:
Fix display to log severity mapping (#83712)

add caplevel to display to pass through
also reverse dict order as 'last update wins'
added tests ... and also log severity to log

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

bcoca committed:
core: raw params better error msg (#83726)

* less confusing error msg
* also remove 'removed' include

s-hertel committed:
csvfile lookup - fix giving an error when no search term is provided (#83710)

Fixes #83689

s-hertel committed:
Fix task-adjacent search path in roles (#83621)

* Restore search path in the current task file’s directory for roles

mattclay committed:
ansible-test - Update venv management and sanity requirements (#83729)

* ansible-test - Update venv management
* Upgrade from pip 24.0 to 24.2
* Omit `wheel` and `setuptools` from ansible-test managed virtual environments
* Drop pre-release hacks
* Update mypy requirements
* Freeze sanity test requirements
* Update sanity test configuration
* Update sanity ignores

mattclay committed:
ansible-test - Update mypy and package-data tests (#83734)

* package-data - Test min/max setuptools version

* Fix multi-version abstraction

* Convert mypy test to script based test

* Fix f-string in pymarkdown test

* Sanity test fixes

nitzmahone committed:
try disabling negative values win_reboot test (#83735)

* hoping to improve CI stability

mattclay committed:
ansible-test - Update default and distro containers (#83736)

mkrizek committed:
Deprecate remaining safe evals (#83293)

s-hertel committed:
Fix csvfile test - quote file argument (#83751)

file was intended to be a string, not an undefined variable

bcoca committed:
Add vaulted_file test (#83717)

* Add vaulted_file test
* fix is_encrypted_file while we are here
Co-authored-by: Martin Krizek <martin.krizek@gmail.com>

bcoca committed:
use diff intermediate var to preserve functionality (#83738)

add tests

akire0ne committed:
fix: `COLOR_INCLUDED` to colorize `included` output (#83711)

adds color included configuration and applies to include message

Co-authored-by: Achille Myette <amyette@drw.com>

kellinm committed:
Enable validation of subkeys in rpm key module (#83716)

* Enable validation of subkeys in rpm key module

A gpg subkey may change while the primary key remains the same. Due to
this behavior, there are situations where validation of the primary gpg
key fingerprint is not sufficient because the desired target is actually
the gpg subkey. This change allows the user to validate against either
the fingerprint of the primary gpg key or its subkey.

Signed-off-by: Kellin <kellin@retromud.org>

* Improve tests, add multi-fingerprint

- Improve tests to cover all cases
- add multi fingerprint validation

Signed-off-by: Kellin <kellin@retromud.org>
# Conflicts:
#	test/integration/targets/rpm_key/tasks/rpm_key.yaml

s-hertel committed:
atomic_move - fix creating file in directory with setgid bit (#83718)

* fix creating file in directory with setgid bit

* add a test using the copy module's content option to create a file in a directory with setgid bit

Co-authored-by: Martin Krizek <martin.krizek@gmail.com>

mattclay committed:
Drop use of setup.py and setup.cfg (#81443)

mattclay committed:
Suppress cryptography warnings for paramiko (#83772)

mkrizek committed:
ini lookup: add new interpolation option (#83773)

Fixes #83755

mkrizek committed:
Fix meta tasks breaking host/fork affinity with host_pinned (#83438)

Fixes #83294

mkrizek committed:
Ensure skipped loop iteration register var is available (#83756)

Fixes #83619

s-hertel committed:
Fix ansible-vault integration test for missing vault ids (#83777)

* Fix broken, circumvented test for missing vault ids

* verify the command returns a non-zero exit code

Co-authored-by: Matt Clay <matt@mystile.com>

mkrizek committed:
Add end_role meta task (#83263)

ci_complete

dkuji committed:
fix copy module update atime/mtime (#83235)

Ensure we force mtime/atime update when using copystat

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>
Co-authored-by: Brian Coca <bcoca@users.noreply.github.com>

mattclay committed:
ansible-test - Remove generation of egg-info (#83786)

Also remove egg-info generation from hacking/env-setup scripts.

mattclay committed:
ansible-test - Add Windows remote connection option

mattclay committed:
ansible-test - Update sanity test requirements (#83795)

* Remove PyYAML pin for yamllint sanity test
* Freeze sanity test requirements

mattclay committed:
ansible-test - Update base/default containers (#83796)

mattclay committed:
Use a venv in more integration tests (#83799)

* Use venv for pause test
* Use venv for debugger test
* Use venv for builtin_vars_prompt test

jborean93 committed:
Fix tests when running against SSH target

jborean93 committed:
Add explicit winrm/psrp tests for HTTP and HTTPS (#83769)

jborean93 committed:
ssh and psrp - Support more complex characters in fetch_file (#83753)

* ssh and psrp - Support more complex chars in fetch_file

Fixes the psrp and ssh (with piped) fetch function to work with paths
that contains glob like characters in the path. For Windows this was
needed when using paths that contain `[]` in the path. For ssh this was
a problem with FreeBSD when using the piped transfer method with similar
characters.

Also tidies up the psrp logic to not inject the paths and buffer size
in the script but pass it as an object through an argument/parameter.

* Fix sanity check

bcoca committed:
Gather mount facts, fallback for when multiproc is not feasable (#83750)

* fallback to 'single threaded gathering' for when multiproc fails

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

bcoca committed:
connection plugins:  extras fix (#83353)

Currently we match the load name, which can be an fqcn, but most users expect the 'naked' name
Now plugins can declare that name by setting _extras_prefix property or fallback to 'non fqcn' if no extras prefix

jborean93 committed:
Fix up raw_params for ansible.windows modules (#83830)

* Fix up raw_params for ansible.windows modules

Fixes up the logic for detecting if using ansible.windows.win_command or
ansible.windows.win_shell with _raw_params. These two modules are
special in that they can be referenced in 4 different ways but the
ansible.windows collection specific prefix needs to be manually added to
the list.

* Fix up sanity issue

mkrizek committed:
dnf: stop filtering exceptions by matching on text (#83297)

* Rely on dnf.base.remove, no special handling isn't needed,
  let the dnf internals figure out what is needed to be done.
  This is more in line with what dnf cli does.

* "already installed" in Exception (if it is even a thing) should be
  caught by special exceptions like MarkingError or CompsError. This
  appears to be a historic check that is no longer needed.

Supersedes: #83295

sivel committed:
Handle authentication errors and token expiration (#83695)

Fixes #70019

jborean93 committed:
powershell - Improve CLIXML parsing (#83847)

Improves the logic used when parsing CLIXML to support all escaped
character sequences and not just newlines.

jctanner committed:
galaxy-cli tasking polling interval from environment variable (#83803)

Added configuration options, including environment variables to control the polling 
No-Issue

---------
Signed-off-by: James Tanner <tanner.jc@gmail.com>
Co-authored-by: s-hertel <19572925+s-hertel@users.noreply.github.com>
Co-authored-by: Jordan Borean <jborean93@gmail.com>

felixfontein committed:
runtime-metadata sanity test: do not fail deprecation version checks if galaxy.yml has empty `version` (#83831)

* Do not create invalid SemanticVersion objects.
* Fix SemanticVersion.parse().
* Add basic runtime-metadata tests.

bcoca committed:
copy, fix permissions and atime on diff partitions (#83824)

we just set time also, when on diff partitions

nitzmahone committed:
fix delegate_to integration test (#83865)

* the test was previously passing erroneously due to the `timeout` elapsing in CI, and that the `failed` test does not encompass `unreachable`

mkrizek committed:
Print the name of the option being deprecated (#83761)

Fixes #83759

mattclay committed:
ansible-test - Update nios-test-container to 5.0.0

jborean93 committed:
Improve testing for Windows SSH and other connection plugins (#83834)

Expands the test matrix used for testing on Windows to cover the three
connection plugins we support for all the tasks. This change also
changes how raw commands are run over SSH to avoid starting a
`powershell.exe` process that was uneeded in the majority of cases used
in Ansible. This simplifies our code a bit more by removing extra
Windows specific actions in the ssh plugin and improves the efficiency
when running tasks.

mattclay committed:
Fix Windows entries in test matrix (#83873)

The display names need to be set so code coverage reporting works.

gundalow committed:
Remove references to IRC or Google Groups (#83875)

mattclay committed:
Update boilerplate sanity test (#83879)

The `annotations` future can now be imported as `_annotations`.

mattclay committed:
ansible-test - Reduce scope of empty-init test (#83878)

jborean93 committed:
Add location on include_tasks fail inside include (#83876)

Adds the datastore details to the parser error when attempting to
include tasks that contain include_tasks without a filename set. This
change will now display the exact location of the include_tasks that
failed like any normal syntax error.

jborean93 committed:
psrp - Remove extras lookups (#83760)

* psrp - Remove extras lookups

Removed the extras variable lookups for the psrp connection plugin. All
valid options are already documented and the extras functionality is
slated to be deprecated at a future point in time. This should have
affect on existing user's playbooks.

* Fix up sanity tests and add explicit boolean conversion test

nitzmahone committed:
Add DaemonThreadPoolExecutor impl (#83880)

* Add DaemonThreadPoolExecutor impl
* Provide a simple parallel execution method with the ability to abandon timed-out operations that won't block threadpool/process shutdown, and without a dependency on /dev/shm (as multiprocessing Thread/Process pools have).
* Create module_utils/_internal to ensure that this is clearly not supported for public consumption.

mattclay committed:
release.py - Add missing setuptools arg to prepare (#83887)

* release.py - Add missing setuptools arg to prepare

This allows the prepare command to accept the `--no-setuptools` argument.

It also fixes a traceback when using the `prepare` command.

* Use a more accurate type hint

mattclay committed:
release.py - Include pyproject.toml in git add (#83892)

bcoca committed:
loop_control "early exit" feature (#62151)

* add a loop_control break_when directive to break out of a loop after any item

* remove loop var as normal exit would

* example usage:

- name: generate a random password up to 10 times, until it matches the policy
  set_fact:
    password: "{{ lookup('password', '/dev/null', chars=character_set, length=length) }}"
  loop: "{{ range(0, 10) }}"
  loop_control:
    break_when:
      - password is match(password_policy)

Co-authored-by: s-hertel <19572925+s-hertel@users.noreply.github.com>

sumantripuraneni committed:
Update subset.yml (#83908)

Typo error in examples

bcoca committed:
delay keyword changed from int to float (#83901)

* delay keyword changed from int to float

* draft test

* fixed test

* expanded test, fixed 'name' tests also

* cleanup

* fix

webknjaz committed:
Stop suggesting `easy_install` in hacking (#83909)

It's been discouraged for the past decade. And CPython actually ships
with pip nowadays, that is bundled within the built-in `ensurepip`
stdlib module.

bcoca committed:
iptables, use existing validation  (#83907)

also remove redundant  and wrong test

infame-io committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

[rpm_key] Add GPG key from a string value

# This is the commit message #2:

[rpm_key] Add GPG key from a string value

Akasurde commented:
@infame-io Are you still working on this? Thanks.
